projects/cm_mcu/CommandLineTask.c

```diff
@@ -932,42 +940,50 @@ static BaseType_t errbuff_in(int argc, char **argv)
 static BaseType_t errbuff_out(int argc, char **argv)
 {
   int copied = 0, s = SCRATCH_SIZE;
-  uint32_t arr[EBUF_NGET];
-  uint32_t (*arrptr)[EBUF_NGET]=&arr;
-  errbuffer_get(ebuf,arrptr);
+  uint32_t num = strtoul(argv[1],NULL,10);
+  uint32_t arr[num];
+  uint32_t (*arrptr)[num]=&arr;
+  errbuffer_get(ebuf,num,arrptr);
 
   copied += snprintf(m+copied, s-copied, "Entries in EEPROM buffer:\r\n");
 
-  int i=0, max=EBUF_NGET;
+  int i=0, max=num;
   while (i<max) {
     uint32_t word = (*arrptr)[i];
 
     uint16_t entry = (uint16_t)word;
     uint16_t errcode = (entry&ERRCODE_MASK)>>ERRDATA_OFFSET;
     uint16_t errdata = entry&ERRDATA_MASK;
     uint16_t counter = entry>>(16-COUNTER_OFFSET);
-    uint16_t realcount = counter*4+1;
+    uint16_t realcount = counter*COUNTER_UPDATE+1;
 
     uint16_t timestamp = (uint16_t)(word>>16);
     uint16_t days = timestamp/0x5a0;
     uint16_t hours = (timestamp%0x5a0)/0x3c;
     uint16_t minutes = timestamp%0x3c;
-    switch(errcode) {
-    case RESTART:
-      copied += snprintf(m+copied, s-copied,
-			 "%02u %02u:%02u \t %x RESTART\r\n", days, hours,
-			 minutes, counter);
-      break;
-    case RESET_BUFFER:
-      copied += snprintf(m+copied, s-copied,
-			 "%02u %02u:%02u \t %x RESET BUFFER\r\n", days,
-			 hours, minutes,counter);
-      break;
-    default:
-      copied += snprintf(m+copied, s-copied,
-			 "%02u %02u:%02u \t %x %x %02x\r\n", days, hours,
-			 minutes, realcount, errcode,errdata);
-      break;
+    if(errcode&(1<<(ERRCODE_OFFSET-1))){
+    	uint8_t status = errcode&((1<<(ERRCODE_OFFSET-1))-1);
+    	copied += snprintf(m+copied, s-copied,
+    				 "%02u %02u:%02u \t %x TEMP %03u %01x\r\n", days, hours,
+    				 minutes, counter, errdata, status);
+    } else{
+		switch(errcode) {
```

Comment:
indentation looks off here

Reply:
Does that look okay now with the new commit, or should I not indent at all for a command that takes up more than one line?